hashes/src/internal_macros.rs

```diff
@@ -73,6 +73,7 @@ macro_rules! hash_trait_impls {
             ///
             /// This is mainly intended as an internal method and you shouldn't need it unless
             /// you're doing something special.
+            #[must_use = "use the returned object to display hex forwards eg., {:x}"]
```

Comment:
These also seem unnecessary -- I have no idea what a user would expect by trying to use this method without the return value, but it seems harmless. But I could go either way.

Reply:
These two are actually pretty nice I think.

Reply:
Two comments:
- I was unsure about most of the messages I put in these macros (i.e., the benefit in having the message)
- I find the attribute noisy but if we are aiming to be a no-footgun library then I rekon we should have these on _all_ pure functions (even constructors where its obvious) because I have made mistakes in the past that were flagged by the compiler because of this attribute

Reply:
I think attributes like this make more sense on methods that have names that might suggest they are mutators while they are actually returning the mutated value. Like `.negate()` on a number or something. For something like `forward_hex` it kinda seems weird. Maybe a user could think that it returns a string and should do `forward_hex().to_string()` instead, but the compiler will tell you that in all cases anyway.